pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -162,13 +184,35 @@ func getV1Registries(config *tomlConfig) ([]Registry, error) {
 // postProcessRegistries checks the consistency of all registries (e.g., set
 // the Prefix to URL if not set) and applies conflict checks.  It returns an
 // array of cleaned registries and error in case of conflicts.
-func postProcessRegistries(regs []Registry) ([]Registry, error) {
-	var registries []Registry
+func postProcessRegistries(regs []Registry) (*SysRegistries, error) {
+	var (
+		registries            []Registry
+		insecureRegistryCIDRs []*net.IPNet
+	)
+	indexConfigs := make(map[string]*IndexInfo)
 	regMap := make(map[string][]Registry)
 
 	for _, reg := range regs {
 		var err error
 
+		// Check if insecure registry is either CIDR notation or of form "host:port"
+		// and add it to either insecureRegistryCIDRs or indexConfigs
+		if reg.Insecure {
+			_, ipnet, err := net.ParseCIDR(reg.URL)
```

Comment:
Cc: @vrothberg who was involved in earlier discussions.

`127.0.0.1/8` is, sadly, a valid namespace or repository name.  It also does not make sense to have the search/mirror functionality associated with CIDR ranges. So, at least in the _new_ format, CIDR configurations should be an explicit, separate component of the config file (i.e. in `tomlConfig`, add a separate `CIDRRanges` top-level field, or something like that — don’t make the semantics of `Registries` ambiguous.

(It’s acceptable for the compatibility code reading v1 to use such a heuristic.)

Reply:
Added a separate component in the tomlConfig for CIDRRanges.